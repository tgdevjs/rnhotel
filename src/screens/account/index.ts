export * from './Account';
export * from './AccountGuest';
export * from './AccountMenu';
export * from './JoinModal';
export * from './SignInModal';
