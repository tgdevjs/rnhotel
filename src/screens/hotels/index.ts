export * from './Hotels';
export * from './ConfirmReservation';
