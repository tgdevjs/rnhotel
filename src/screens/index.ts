export * from './account';
export * from './hotels';
export * from './search';
export * from './stays';
