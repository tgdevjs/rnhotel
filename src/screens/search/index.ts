export * from './Search';
export * from './SearchCalendar';
