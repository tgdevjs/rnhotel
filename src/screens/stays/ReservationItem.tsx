import React from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import Swipeout from "react-native-swipeout";
import moment from "moment";

import { DateRange, HotelTitle, DateRangeError } from "../../components";
import { ReservationType } from "../../types";
import { colors } from "../../styles";

type Props = {
  item: ReservationType;
  onDeleteItem: (id: string) => void;
};

const styles = StyleSheet.create({
  header: {
    alignItems: "center",
    borderColor: colors.gray,
    borderWidth: 1,
    borderRadius: 10,
    justifyContent: "center",
    marginHorizontal: 10,
    height: 160
  },
  textContainer: {
    paddingLeft: 42
  },
  dateRange: {
    paddingBottom: 10
  },
  dateText: {
    fontSize: 30,
    fontWeight: "bold"
  },
  hotelTitle: {
    color: colors.black
  },
  name: {
    fontSize: 16,
    fontWeight: "bold"
  },
  date: {
    flexDirection: "row"
  }
});

export const ReservationItem = ({
  item: { arrivalDate, departureDate, hotelName, id },
  onDeleteItem
}: Props) => {
  return (
    <Swipeout
      right={[
        {
          text: "Delete",
          backgroundColor: colors.red,
          onPress: () => onDeleteItem(id)
        }
      ]}
      autoClose
      backgroundColor="transparent"
    >
      <TouchableOpacity onPress={() => {}}>
        <View style={styles.header}>
          {!moment(arrivalDate).isValid() ||
          !moment(departureDate).isValid() ? (
            <DateRangeError startDay={arrivalDate} endDay={departureDate} />
          ) : (
            <DateRange
              containerStyle={styles.dateRange}
              dateTextStyle={styles.dateText}
              startDay={arrivalDate}
              endDay={departureDate}
            />
          )}
          <HotelTitle style={styles.hotelTitle} title={hotelName} />
          <Text>{id}</Text>
        </View>
      </TouchableOpacity>
    </Swipeout>
  );
};
