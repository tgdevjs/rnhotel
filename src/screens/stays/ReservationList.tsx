import React from "react";
import { Query, Mutation } from "react-apollo";
import { FlatList, StyleSheet, Text, View } from "react-native";

import { ReservationItem } from ".";
import { ReservationType } from "../../types";
import {
  reservationsQuery,
  deleteReservationMutation
} from "../../apollo/queries";
import { getReservationsQueryVariables } from "../../utils";
import { fonts } from "../../styles";

type Props = {
  name: string;
  orderBy: string;
  timeRangeKey: string;
};

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  itemSeparator: {
    height: 8
  },
  listEmpty: {
    alignItems: "center",
    flex: 1,
    height: 400,
    justifyContent: "center"
  },
  actionText: fonts.H3
});

export const ReservationList = ({ name, orderBy, timeRangeKey }: Props) => {
  const variables = getReservationsQueryVariables({
    name,
    orderBy,
    timeRangeKey
  });
  const timeRange = timeRangeKey === "arrivalDate_gte" ? "upcoming" : "past";

  return (
    <Mutation
      mutation={deleteReservationMutation}
      update={(cache, { data: { deleteReservation } }) => {
        const { reservations } = cache.readQuery({
          query: reservationsQuery,
          variables
        });
        cache.writeQuery({
          query: reservationsQuery,
          variables,
          data: {
            reservations: reservations.filter(
              (item: ReservationType) => item.id !== deleteReservation.id
            )
          }
        });
      }}
    >
      {deleteReservation => (
        <Query query={reservationsQuery} variables={variables}>
          {({ data, loading }) => {
            if (loading) return null;

            return (
              <View style={styles.container}>
                <FlatList
                  data={data.reservations}
                  horizontal={false}
                  ItemSeparatorComponent={() => (
                    <View style={styles.itemSeparator} />
                  )}
                  keyExtractor={(item: ReservationType) => item.id}
                  ListEmptyComponent={
                    <View style={styles.listEmpty}>
                      <Text style={styles.actionText}>
                        {`No ${timeRange} stays`}
                      </Text>
                    </View>
                  }
                  renderItem={({ item }) => (
                    <ReservationItem
                      item={item}
                      onDeleteItem={id =>
                        deleteReservation({ variables: { where: { id } } })
                      }
                    />
                  )}
                />
              </View>
            );
          }}
        </Query>
      )}
    </Mutation>
  );
};
