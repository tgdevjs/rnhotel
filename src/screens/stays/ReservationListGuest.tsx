import React from "react";
import { StyleSheet, Text, View } from "react-native";

import { SignInButton } from "../../components";

type Props = {
  onSignIn: () => void;
};

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  main: {
    alignItems: "center"
  },
  actionSection: {
    margin: 20,
    alignItems: "center"
  },
  actionText: {
    fontSize: 20,
    fontWeight: "bold"
  },
  valueText: {
    fontSize: 20
  }
});

export const ReservationListGuest = ({ onSignIn }: Props) => {
  return (
    <View style={styles.mainContainer}>
      <View style={styles.main}>
        <View style={styles.actionSection}>
          <Text style={styles.actionText}>Sign in</Text>
          <Text style={styles.valueText}>to see your reservations</Text>
        </View>
        <SignInButton onPress={onSignIn} />
      </View>
    </View>
  );
};
