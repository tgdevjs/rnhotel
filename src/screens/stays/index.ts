export * from './Stays';
export * from './StayItem';
export * from './StaysList';
export * from './StaysListGuest';
