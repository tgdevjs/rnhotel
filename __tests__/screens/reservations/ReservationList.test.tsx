import React from "react";
import { ReservationList } from "../../../src/screens/stays";

describe("ReservationList", () => {
  it("renders default", () => {
    expect(shallow(<ReservationList />)).toMatchSnapshot();
  });
});
