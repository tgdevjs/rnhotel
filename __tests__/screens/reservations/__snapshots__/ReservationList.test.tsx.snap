// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationList renders default 1`] = `
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "deleteReservation",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "where",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "where",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "deleteReservation",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "hotelName",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "arrivalDate",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "departureDate",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ReservationWhereUniqueInput",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "where",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 193,
        "start": 0,
      },
    }
  }
  update={[Function]}
>
  <Component />
</Mutation>
`;
